### PR TITLE
New patching

### PR DESCRIPTION
**Pull request by matthewalbani**

### CODE REVIEW

**Comment by matthewalbani** [comment]
goes out after solanolabs/tddium_system#688

**Comment by matthewalbani** [comment]
@wkj  PTAL

File: lib/solano/constant.rb [hunk 1]
```diff
@@ -102,6 +102,10 @@ module Warning
  depending on how large your repo is. Rest assured, you'll only need to run hg:mirror once.)
 
 EOF
+      GIT_VERSION_FOR_PATCH =<<EOF
+Patching require a newer version of git. Please update to git 1.8+ or contact support at support@solanolabs.com
```
Comment (wkj) [inline lib/solano/constant.rb]: Edit nit: Patching require*s*.

File: lib/solano/scm/git.rb [hunk 2]
```diff
@@ -143,13 +153,7 @@ def create_snapshot(session_id, options={})
       if !options[:force] then
         #git default branch
         branch = options[:default_branch]
-        branches = /\-\>.*\/(.*)$/.match( (`git branch -r | grep origin/HEAD`).strip)
```
Comment (wkj) [inline lib/solano/scm/git.rb]: For my own edification: what did this code do and why is it now gone?

File: lib/solano/scm/git.rb [hunk 3]
```diff
@@ -187,23 +191,47 @@ def create_snapshot(session_id, options={})
     end
 
     def create_patch(session_id, options={})
+      if !check_version('1.7.12.4') then
```
Comment (wkj) [inline lib/solano/scm/git.rb]: Seems very specific.  How did we choose this version?

Comment (matthewalbani) [inline lib/solano/scm/git.rb]: its the oldest version that we support on workers, and therefore the oldest version its been tested on. The messaging actually asks them to update to 1.8+

File: lib/solano/scm/git.rb (hunk 2 above)
Comment (matthewalbani) [inline lib/solano/scm/git.rb]: it was meant to try and figure out what the default branch was. replaces with the (default_branch function)[https://github.com/solanolabs/solano/blob/master/lib/solano/scm/git.rb#L68] which i had not noticed when i wrote the above

Comment (wkj) [inline lib/solano/scm/git.rb]: 👍

File: lib/solano/scm/git.rb (hunk 3 above)
Comment (wkj) [inline lib/solano/scm/git.rb]: Maybe add a comment? E.g. ``oldest supported version of git``?  Pedantic, I know.

Comment (matthewalbani) [inline lib/solano/scm/git.rb]: added

**Pull request merged**